sdks/java/core/src/test/java/org/apache/beam/sdk/schemas/utils/AvroUtilsTest.java

```diff
@@ -324,6 +339,59 @@ public void testFromBeamSchema() {
     assertEquals(getAvroSchema(), avroSchema);
   }
 
+  @Test
+  public void testAvroSchemaFromBeamSchemaCanBeParsed() {
+    String stringSchema = AvroUtils.toAvroSchema(getBeamSchema()).toString();
+    org.apache.avro.Schema schema = new org.apache.avro.Schema.Parser().parse(stringSchema);
+    assertEquals(stringSchema, schema.toString());
```

Comment:
Why do we compare JSON representation instead of actual objects? Should we change the rest of the comparisons as well?

Reply:
Very light preference and habit for comparing Avro schemas as a string, so the junit error messages show the diff on failure!  I have no objection to changing it!

Reply:
I don't have a preference except that it should be consistent, we can extract it as a separate matcher if having good error messages is something that we want.

Reply:
Changed and rebased with the recent fix on master!

I interactively squashed the commits on my end to retain @iemejia's work.